op-proposer/proposer/driver.go

```diff
@@ -77,15 +77,19 @@ type L2OutputSubmitter struct {
 }
 
 // NewL2OutputSubmitter creates a new L2 Output Submitter
-func NewL2OutputSubmitter(setup DriverSetup) (*L2OutputSubmitter, error) {
+func NewL2OutputSubmitter(setup DriverSetup) (_ *L2OutputSubmitter, err error) {
 	ctx, cancel := context.WithCancel(context.Background())
+	defer func() {
+		if err != nil || recover() != nil {
+			cancel()
+		}
+	}()
```

Comment:
It's fine to cancel a second time / cancel an anyway GC'ed context, and the `defer` will run even when there's a panic (hence the recover being in here). I don't think it makes sense to recover from panics here either, since it's just a constructor, and a panic would be a programmer error, not some user-initiated one-off action.

```suggestion
	defer cancel()
```

The real issue here is that there shouldn't be fetching / network-calls / dynamic error-cases in constructors.

Reply:
Or is this context long-lived, for the duration of the output-submitter lifetime? If so, it shouldn't be, the context should then be managed within the output-submitter, and cancelled with a `Close()` call on the output submitter.

Reply:
Yes, this context is long-lived, and passed into the `L2OutputSubmitter`, which is why we don't `defer cancel()` here. This behavior isn't new, and the `L2OutputSubmitter` cancels this context in `StopL2OutputSubmitting`.

This should only cancel the context early if an error was returned by `NewL2OutputSubmitter` or `NewL2OutputSubmitter` panicked

Reply:
https://github.com/ethereum-optimism/client-pod/issues/443